internal/dag/gatewayapi_processor_test.go

```diff
@@ -199,20 +199,20 @@ func TestComputeHosts(t *testing.T) {
 			want:         sets.NewString("http.projectcontour.io"),
 			wantError:    nil,
 		},
-		"listener host with many labels doesn't match hostnames wildcard host": {
-			listenerHost: "too.many.labels.projectcontour.io",
+		"listener host with many labels matches hostnames wildcard host": {
+			listenerHost: "very.many.labels.projectcontour.io",
 			hostnames: []gatewayapi_v1alpha2.Hostname{
 				"*.projectcontour.io",
 			},
-			want:      nil,
+			want:      sets.NewString("very.many.labels.projectcontour.io"),
 			wantError: nil,
 		},
 		"listener wildcard host doesn't match hostnames with many labels host": {
```

Comment:
should this test case description be updated? "listener wildcard host matches hostnames with many labels host"

Reply:
fixed!